app/settings.go

```diff
@@ -170,7 +168,13 @@ func (s *settings) setupTheme() {
 		variant = theme.VariantDark
 	}
 
-	s.applyTheme(effectiveTheme, variant)
+	if async.IsMainGoroutine() {
```

Comment:
Is it OK that setupTheme is sometimes called from non-main?

Reply:
That was the reasoning behind adding the IsMainGoroutine check. setupTheme is called from platform-specific code and I wasn't sure which of those would be on main, and which may be from a goroutine

Reply:
I don't think we should rely too much on `async.IsMainGoroutine()`. There is a reason that the Go developers don't expose the goroutine ID in a simple way as it allows people to more easily write bad code. Can we not use `async.EnsureMain()` for now until the migration is complete?

Reply:
All OS specific code should come in on main. If it doesn't then other things will break.

Reply:
Then this should not do the ID check at all, right?

Reply:
Cool. Agreed :)